crates/modelardb_common/src/storage.rs

```diff
@@ -330,6 +335,34 @@ pub async fn extract_azure_blob_storage_arguments(
     Ok((account, access_key, container_name, offset_data))
 }
 
+/// Reinterpret the bits used for univariate ids in `compressed_segments` to convert the column from
+/// [`UInt64Array`] to [`Int64Array`] as the Delta Lake Protocol does not support unsigned integers.
+fn univariate_ids_uint64_to_int64(compressed_segments: &mut Vec<RecordBatch>) {
+    for record_batch in compressed_segments {
+        let mut columns = record_batch.columns().to_vec();
+        let univariate_ids = crate::array!(record_batch, 0, UInt64Array);
+        let signed_univariate_ids: Int64Array =
+            univariate_ids.unary(|value| i64::from_ne_bytes(value.to_ne_bytes()));
```

Comment:
is it safe to use native endianness? Do we risk to write with one endianness and then later read with another endianess (on the client)?

Reply:
Attempts have been made to ensure ModelarDB works on systems with different endianess, e.g., the [compression purposely uses little-endian](https://github.com/ModelarData/ModelarDB-RS/blob/84c886f07dbd470ff8a5803ca68dee3ea246fe03/crates/modelardb_compression/src/types.rs#L298) to ensure a specific endianess is used for compressed segments regardless of the system's endianess. Little-endiann was chosen as high performance CPUs such as [X86_64 and M-series](https://developer.apple.com/documentation/apple-silicon/porting-your-macos-apps-to-apple-silicon#Address-Architectural-Differences) uses little-endian. Unfortunately, GitHub Actions does not, as far as I am aware, [provide runners for any big-endian systems](https://github.com/actions/runner-images). Thus, ModelarDB has never been tested on anything but little-endian, so I am certain that there are issues we have not considered, e.g., Apache Arrow specifically states that it is [little-endian by default](https://arrow.apache.org/docs/format/Columnar.html#byte-order-endianness).

Regarding the use of native-endianness in the function executed by `unary()`, it was purposely used as [`unary()`](https://docs.rs/arrow/latest/arrow/array/struct.PrimitiveArray.html#method.unary) operates on [`ArrowPrimitiveType`](https://docs.rs/arrow/latest/arrow/datatypes/trait.ArrowPrimitiveType.html) which in this case is [`UInt64Type`](https://docs.rs/arrow/latest/arrow/datatypes/struct.UInt64Type.html) which corresponds to the native Rust type [`u64`](https://doc.rust-lang.org/nightly/std/primitive.u64.html) which is native-endian, e.g., see the documentation for [from_be()](https://doc.rust-lang.org/nightly/std/primitive.u64.html#method.from_be). However, according to the [documentation for Apache Parquet](https://parquet.apache.org/docs/file-format/data-pages/encodings/), it always stores integers as little-endian. So if I understand the process correctly, integers are stored in Apache Parquet as little-endian, read into and stored in Apache Arrow as little-endian, converted to native-endian when manipulated by `unary()` which is a [no-op for little-endian systems](https://doc.rust-lang.org/nightly/std/primitive.u64.html#method.from_le), and than written back to Apache Arrow as little-endian. Thus, the function executed by `unary()` must use native-endianess to be portable. However, there may certainly be something that I have overlooked or misunderstood, and as stated above, the code has only been tested on little-endian systems as I do not have access to systems with any other endianess.